### PR TITLE
[front] Add targets view to Localisation/Entities/Vulnerability to get all the targeting entities + Filters

### LINKED ISSUE #2473: Be able to list threats targeting a specific country and sector (with advanced filters)
## Use case

Be able to list threats targeting a specific country and sector (with advanced filters)

### PR DESCRIPTION
**Pull request by Kedae**

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js [hunk 1]
```diff
@@ -64,6 +64,28 @@ class VulnerabilityKnowledgeComponent extends Component {
...
+            path="/dashboard/arsenal/vulnerabilities/:vulnerabilityId/knowledge/targets"
+            render={(routeProps) => (
+              <EntityStixCoreRelationships
+                entityId={vulnerability.id}
+                relationshipTypes={['targets']}
+                isRelationReversed
+                entityLink={link}
+                allDirections={true}
+                stixCoreObjectTypes={[
+                  'Threat-Actor',
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js:78]: According to the stix.ts file, it's missing the type Attack-Pattern.

File: opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectKnowledgeBar.js [hunk 2]
```diff
@@ -80,6 +80,18 @@ class StixCoreObjectKnowledgeBar extends Component {
             </ListItemIcon>
             <ListItemText primary={t('Overview')} />
           </MenuItem>
+          <MenuItem
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectKnowledgeBar.js]: This menu is always available now ?
We have blank screen in some entity.
image

File: opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js [hunk 3]
```diff
@@ -63,6 +63,28 @@ class EventKnowledgeComponent extends Component {
...
+            exact
+            path="/dashboard/entities/events/:eventId/knowledge/targets"
+            render={(routeProps) => (
+              <EntityStixCoreRelationships
+                entityId={event.id}
+                relationshipTypes={['targets']}
+                isRelationReversed
+                entityLink={link}
+                allDirections={true}
+                stixCoreObjectTypes={[
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js:75]: According to the stix.ts file, the campaign entity is not available for an event

File: opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualKnowledge.js [hunk 4]
```diff
@@ -77,6 +77,28 @@ class IndividualKnowledgeComponent extends Component {
...
+            exact
+            path="/dashboard/entities/individuals/:individualId/knowledge/targets"
+            render={(routeProps) => (
+              <EntityStixCoreRelationships
+                entityId={individual.id}
+                relationshipTypes={['targets']}
+                isRelationReversed
+                entityLink={link}
+                allDirections={true}
+                stixCoreObjectTypes={[
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualKnowledge.js:89]: According to the stix.ts file, it's missing the type Attack-Pattern and the campaign type is not available.

File: opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationKnowledge.js [hunk 5]
```diff
@@ -77,6 +77,28 @@ class OrganizationKnowledgeComponent extends Component {
...
+            exact
+            path="/dashboard/entities/organizations/:organizationId/knowledge/targets"
+            render={(routeProps) => (
+              <EntityStixCoreRelationships
+                entityId={organization.id}
+                relationshipTypes={['targets']}
+                isRelationReversed
+                entityLink={link}
+                allDirections={true}
+                stixCoreObjectTypes={[
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationKnowledge.js:89]: According to the stix.ts file, it's missing the type Attack-Pattern.

File: opencti-platform/opencti-front/src/private/components/entities/sectors/SectorKnowledge.js [hunk 6]
```diff
@@ -64,6 +64,28 @@ class SectorKnowledgeComponent extends Component {
...
+            exact
+            path="/dashboard/entities/sectors/:sectorId/knowledge/targets"
+            render={(routeProps) => (
+              <EntityStixCoreRelationships
+                entityId={sector.id}
+                relationshipTypes={['targets']}
+                isRelationReversed
+                entityLink={link}
+                allDirections={true}
+                stixCoreObjectTypes={[
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/entities/sectors/SectorKnowledge.js:76]: According to the stix.ts file, it's missing the type Attack-Pattern.

File: opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js [hunk 7]
```diff
@@ -78,6 +78,28 @@ class SystemKnowledgeComponent extends Component {
...
+            exact
+            path="/dashboard/entities/systems/:systemId/knowledge/targets"
+            render={(routeProps) => (
+              <EntityStixCoreRelationships
+                entityId={system.id}
+                relationshipTypes={['targets']}
+                isRelationReversed
+                entityLink={link}
+                allDirections={true}
+                stixCoreObjectTypes={[
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js:90]: According to the stix.ts file, the tool entity is not available.

File: opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js (hunk 1 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js:78]: Maybe it's better to don't pass any entities ? Same for the next files.

File: opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js (hunk 3 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js:76]: Maybe it's better to don't pass any entities ?

File: opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualKnowledge.js (hunk 4 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualKnowledge.js:90]: Maybe it's better to don't pass any entities ?

**Comment by RomuDeuxfois** [comment]
About the filter, we can target any entities but just some of them are available. Maybe it's better to filter on available entities ?
image

I'm not sure the list on the target entity filter is correct.
image

Now, the target filter is available in all the knowledge menus.

File: opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js:78]: May be you have a point about Attack-Pattern I'll check with Samuel, the list comes from him and we don't want anything else (to answer your second question)

File: opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectKnowledgeBar.js (hunk 2 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectKnowledgeBar.js]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js (hunk 3 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js:75]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js (hunk 7 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js:90]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/sectors/SectorKnowledge.js (hunk 6 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/sectors/SectorKnowledge.js:76]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationKnowledge.js (hunk 5 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationKnowledge.js:89]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualKnowledge.js (hunk 4 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualKnowledge.js:89]: Fixed

File: opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityKnowledge.js:78]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js (hunk 3 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js:75]: The list still not aligned with the stix.ts file. Maybe some relationship is missing in this file.

Existing
ENTITY_TYPE_INCIDENT
ENTITY_TYPE_INTRUSION_SET
ENTITY_TYPE_MALWARE
ENTITY_TYPE_THREAT_ACTOR
ENTITY_TYPE_TOOL

Your adding
PR ENTITY_TYPE_CAMPAIGN

File: opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js (hunk 7 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js:90]: The list still not aligned with the stix.ts file. Maybe some relationship is missing in this file.

Existing
ENTITY_TYPE_CAMPAIGN
ENTITY_TYPE_INCIDENT
ENTITY_TYPE_INTRUSION_SET
ENTITY_TYPE_MALWARE
ENTITY_TYPE_THREAT_ACTOR

Your adding
ENTITY_TYPE_TOOL

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/systems/SystemKnowledge.js:90]: FIxed

File: opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js (hunk 3 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/entities/events/EventKnowledge.js:75]: Fixed

**Pull request merged**